examples/forms/components/Forms/ErrorMessages.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { useFormContext } from "react-hook-form";
+import FieldContext from "./FieldContext";
+
+export type ErrorMessagesProps = {
+  pattern?: string;
+  required?: string;
+  maxLength?: string;
+  [type: string]: string;
+};
+
+export const Error = (props: React.ComponentProps<"div">) => (
+  <div className="form-error" role="alert" {...props} />
+);
+
+const ErrorMessages = ({
+  pattern = "Toto pole má nesprávny formát",
+  required = "Toto pole je povinný údaj",
+  maxLength = "Presiahli ste maxímálnu dĺžku poľa",
+  ...types
+}: ErrorMessagesProps) => {
+  const { id } = React.useContext(FieldContext);
+  const {
+    formState: { errors },
+  } = useFormContext();
+
+  return (
+    <>
+      {errors?.[id]?.type === "pattern" && <Error>{pattern}</Error>}
```

Comment:
Toto by imho bolo treba prerobit. Pretoze idecko je vo formulari len kvoli tomu, aby bol spojeny label s fieldom. Ziadna ina logika by sa nanho nemalo viazat, lebo musi byt unikatne pre kazdy jeden field (co napr. v Slnecniciach nie je a prinasa to problem s tym, ked chces mat rovnaky formular v modali a na stranke naraz). V react 18 na to bude hook [useId](https://github.com/reactwg/react-18/discussions/111), ktory prinesie stabilne generovane idecka aj do SSR.

Spravne pole pre identifikaciu pola je `name` a tak aj react hook forms funguje a v `errors` objekte su error organizovane podla `name`, ktore je totozne s prvym parametrom funkcie [register](https://react-hook-form.com/api/useform/register).

Reply:
aktualizoval som aby sa používal name miesto idčka.

Reply:
<3